src/models/stores/class.ts

```diff
@@ -1,44 +1,60 @@
 import { types } from "mobx-state-tree";
 import { ClassInfo } from "../../lib/auth";
+import { values } from "lodash";
 
-export const ClassStudentModel = types
-  .model("ClassStudent", {
+export enum ClassUserType {
+  teacher = "teacher",
+  student = "student"
+}
+
+export const ClassUserModel = types
+  .model("ClassUser", {
+    type: types.enumeration<ClassUserType>("ClassUser", values(ClassUserType) as ClassUserType[]),
     id: types.string,
     firstName: types.string,
     lastName: types.string,
     fullName: types.string,
     initials: types.string,
+  })
+  .views((self) => {
+    return {
+      get displayName() {
+        return self.type === ClassUserType.student
```

Comment:
I would write the test the other way. If a new user type were added the full name would be more likely to be correct than the teacher version, so it would be the better "default".

Reply:
Good point